libs/MeshKernel/src/Mesh2D.cpp

```diff
@@ -2236,20 +2236,24 @@ std::unique_ptr<Mesh2D> Mesh2D::Merge(const Mesh2D& mesh1, const Mesh2D& mesh2)
 
 meshkernel::BoundingBox Mesh2D::GetBoundingBox() const
 {
-
     Point lowerLeft(std::numeric_limits<double>::max(), std::numeric_limits<double>::max());
-    Point upperRight = -lowerLeft;
+    Point upperRight(std::numeric_limits<double>::lowest(), std::numeric_limits<double>::lowest());
 
     const auto numNodes = GetNumNodes();
-    for (UInt e = 0; e < numNodes; ++e)
+    for (UInt n = 0; n < numNodes; ++n)
     {
-        lowerLeft.x = std::min(lowerLeft.x, m_nodes[e].x);
-        lowerLeft.y = std::min(lowerLeft.y, m_nodes[e].y);
-        upperRight.x = std::max(upperRight.x, m_nodes[e].x);
-        upperRight.y = std::max(upperRight.y, m_nodes[e].y);
+        if (!m_nodes[n].IsValid())
+        {
+            continue;
+        }
```

Comment:
Do you think we should using [[likely]] and [[unlikely]] attributes in some cases?